### PR TITLE
Allowing the system to check that the date of birth on the SA ID number is correct

### PR DESCRIPTION
**Pull request by sewagodimo**

### CODE REVIEW

File: go-app.js [hunk 1]
```diff
@@ -45,7 +45,18 @@ go.utils = {
             sum += +c;
         }
         sum = 10 - ('' + sum).charAt(1);
-        return ('' + sum).slice(-1) == check;
+        if( ('' + sum).slice(-1) == check){
+            //the id_no is valid. NOw check that valid will accept the date
+            var month,day,year;
```
Comment (miltontony) [inline go-app.js]: You can declare these variables inline with the assignment
```javascript
var year = parseInt(id.substr(0,2));
```

File: go-app.js [hunk 2]
```diff
@@ -45,7 +45,18 @@ go.utils = {
...
+            year = parseInt( id.substr(0,2));
+            month =parseInt( id.substr(2,2));
```
Comment (miltontony) [inline go-app.js]: please watch your spacing and be consistent. `var month = parseInt(id.substr(2,2));` is better.

File: test/app.test.js [hunk 3]
```diff
@@ -455,7 +466,7 @@ describe("app", function() {
         });
 
         describe('Submit_Case', function () {
-            it('should make a valid Ona API call', function () {
+            it('should make a valid Ona API call1', function () {
```
Comment (miltontony) [inline test/app.test.js]: typo?

File: test/app.test.js [hunk 4]
```diff
@@ -367,14 +367,25 @@ describe("app", function() {
...
+
 
             it('should accept valid SA ID numbers', function () {
                 return tester
                     .setup.user.state('SA_ID_Entry')
                     .setup.user.answers({
                         'Facility_Code_Entry': '111111'
                     })
-                    .input('8905100273087')
+                    .input('9602294800085')
```
Comment (miltontony) [inline test/app.test.js]: What's the significance of changing this? If it serves no purpose, you can just leave it as is.

File: test/app.test.js [hunk 5]
```diff
@@ -22,7 +22,7 @@ describe("app", function() {
             Locality_Entry: 'Locality_Entry',
             Locality_Entry_Other: 'Locality_Entry_Other',
             MSISDN_Entry: 'MSISDN_Entry',
-            date_of_birth: '1980-01-02',
+            date_of_birth: '1980-02-21',
```
Comment (miltontony) [inline test/app.test.js]: Do these dates need to change? Can't we just leave them as is?

**Review by miltontony — CHANGES_REQUESTED** [review]
The changes look good, just some style fixes here and there 🍰

File: go-app.js [hunk 6]
```diff
@@ -45,7 +45,16 @@ go.utils = {
             sum += +c;
         }
         sum = 10 - ('' + sum).charAt(1);
-        return ('' + sum).slice(-1) == check;
+        if( ('' + sum).slice(-1) == check){
```
Comment (miltontony) [inline go-app.js]: Any reason why there's a space between the first 2 brackets?

File: go-app.js [hunk 7]
```diff
@@ -45,7 +45,16 @@ go.utils = {
...
+            //the id_no is valid. Then check that valid will accept the date
+            var year = parseInt( id.substr(0,2));
+            var month = parseInt( id.substr(2,2));
```
Comment (miltontony) [inline go-app.js]: similarly here - please remove the space after the brackets

**Comment by sewagodimo** [comment]
@miltontony fixed the formatting errors and ran the test on them

**Review by miltontony — APPROVED** [review]
👍 🍰 wtih

**Pull request merged**